Code/Framework/AzCore/AzCore/Name/Name.cpp

```diff
@@ -187,7 +333,7 @@ namespace AZ
                 ->Method("ToString", &Name::GetCStr)
                 ->Method("Set", &Name::SetName)
                 ->Method("IsEmpty", &Name::IsEmpty)
-                ->Method("Equal", &Name::operator==)
+                ->Method("Equal", static_cast<bool(Name::*)(const Name&)const>(&Name::operator==))
```

Comment:
You can use a lambda for this
```suggestion
                ->Method("Equal", [](const Name* lhs, const Name& rhs) { return *lhs == rhs; })
```

But this is fine as is, if you prefer this style.

Reply:
I think I'll leave this as-is, but good to know that BC supports lambdas, that'll be helpful.